lib/msprime.c

```diff
@@ -5019,6 +5019,7 @@ msp_run_sweep(msp_t *self)
         /* convert time scale */
         pop_size = get_population_size(&self->populations[0], self->time);
         self->time = t_start + (time[curr_step - 1] * self->ploidy * pop_size);
```

Comment:
no the time steps should never be zero size. i.e. https://github.com/tskit-dev/msprime/blob/bcb0b46db15a81550c85fc4472ba7681d923c0db/lib/msprime.c#L6974

I'm just adding a tiny epsilon to the beginning of the sweep phase because the problem seemed to be zero length branches being added at only the first possible event during the sweep phase. this is equivalent to a wait time of zero being returned from an exponential wait time.

Reply:
Try adding the assert above - it tripped for me.

Reply:
so that's the time, and not the dt. that will trip as zero on the first iteration of the loop, as the time[0] = 0  i believe, which is why I've added epsilon to t_start

Reply:
now if this trips when curr_step > 1 there is an issue

Reply:
What you need to do to check is to add
```
tsk_bug_assert(t + trajectory.dt > t);
```
to https://github.com/tskit-dev/msprime/blob/bcb0b46db15a81550c85fc4472ba7681d923c0db/lib/msprime.c#L6974  right?

Reply:
i mean i could, but that has to be the case because trajectory.dt is always > 1?

Reply:
OK, my bad - but can you add in a bunch of bug_asserts here that do make sense please @andrewkern? I want to understand which term of this ``(time[curr_step - 1] * self->ploidy * pop_size);`` is zero.

We need to understand the actual problem here, numerical bugs are absolutely evil and they *will* pop back up unless dealt with at source.

Reply:
yeah can do. 

another way to fix this bug would be if `time[curr_step - 1] == 0` (and only in that case) add epsilon. 

any preference to that solution over just shifting everything in this phase by epsilon, as i'm currently doing?